src/routes/donors.js

```diff
@@ -1,155 +1,158 @@
-const express = require('express')
-const router = express.Router()
-const Sequelize = require('sequelize')
-const db = require('../models/index')
-const pagination = require('../middlewares/pagination')
-const _ = require('lodash')
-const summation = require('../lib/math')
-
+const express = require("express");
+const router = express.Router();
+const Sequelize = require("sequelize");
+const db = require("../models/index");
+const pagination = require("../middlewares/pagination");
+const _ = require("lodash");
+const summation = require("../lib/math");
+const BigNumber = require("bignumber.js");
 //donor list table
-router.get('/', pagination, function (req, res, next) {
-  let offset = req.customParams.offset
-  let limit = req.customParams.limit
-
-  var donor = db.Donor
-  const now = new Date()
-  const thisYear = now.getFullYear()
-  const thisMonth = now.getMonth()
-  const thisDate = now.getDate()
-  // const {
-  //   from = new Date(thisYear, 0, 1).toISOString(),
-  //   to = new Date(thisYear, thisMonth, thisDate).toISOString(),
-  //   page = 1
-  // } = req.query;
+router.get("/", function (req, res, next) {
+  const { from, to, taxDeduc, minAmt, maxAmt, source } = req.query;
+  const donationConditions = {};
+  const sourceConditions = {};
 
-  //   try {
-  //     const donors = await db.Donor.findAll({
-  //       limit: limit,
-  //       offset: offset,
-  //       order: [["id", "ASC"]]
-  //     });
-  //     res.json({
-  //       data: donors,
-  //       perPage: limit,
-  //       offset: offset
-  //     });
-  //   } catch (err) {
-  //     console.log(err);
-  //     res.status(500).json(err);
-  //   }
-  // });
+  const minAmtNum = minAmt ? BigNumber(minAmt) : BigNumber(0);
+  const maxAmtNum = maxAmt ? BigNumber(maxAmt) : BigNumber(Infinity);
+  if (minAmtNum.isNaN() || maxAmtNum.isNaN()) {
+    return res.status(422).json({
+      message: "minAmt and/ or maxAmt is not a number",
+    });
+  }
+  if (minAmtNum.isGreaterThan(maxAmtNum)) {
+    return res.status(422).json({
+      message: "minAmt is more than maxAmt"
+    })
+  }
 
-  try {
-    donor
-      .findAll({
-        limit: limit,
-        offset: offset,
-        // where (for advanced filters)
-        attributes: [
-          'idNo',
-          'name',
-          'contactNo',
-          'email',
-          'dnc',
-          [
-            Sequelize.fn('SUM', Sequelize.col('donationAmount')),
-            'totalAmountDonated',
-          ],
-        ],
+  if (from && to) {
+    donationConditions.donationDate = {
+      [Sequelize.Op.between]: [new Date(from), new Date(to)],
+    };
+  } else {
```

Comment:
If the FE query doesn't include a date, then it should show all donors by default I think

Reply:
Hi @SheldonCheng , I was referring to thomas's comment in trello https://trello.com/c/SDG6dNjQ/43-donor-list-page-basic-filters

and was following up w kaleng, she told me the same that the default donor list page will always be first day of the year to today's date. Then its all up to the user to change the date filters if they want to see other date range